### PR TITLE
Feature/Enemy ragdoll

### PR DESCRIPTION
**Pull request by rafaelgonzalez**
- Added enemy ragdoll prefab
- Enemy spawns ragdoll when killed (transform positions are copied over through script)
- Enemy ragdoll turns black and eye glow fades over time
- Enemy ragdolls are buried underground and destroyed after a few seconds
- Apply bullet velocity as force on contact point with ragdoll

### CODE REVIEW

File: Assets/Scripts/ChangeMaterialColorOverTime.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ChangeMaterialColorOverTime : MonoBehaviour {
```
Comment (LaurieTrichet) [inline Assets/Scripts/ChangeMaterialColorOverTime.cs]: You should use an animation clip on the enemy ragdoll game object to blacken the shaders.

File: Assets/Scripts/DestroyOnCollision.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using UnityEngine;
+
+public class DestroyOnCollision : MonoBehaviour {
+	
+	public bool destroySelf;
+	public bool destroyCollidingObject;
+
+	void OnCollisionEnter (Collision collision) {
+		if (destroyCollidingObject == true) {
```
Comment (LaurieTrichet) [inline Assets/Scripts/DestroyOnCollision.cs]: This should live in the collision code on the enemy side.

File: Assets/Scripts/EnemyBulletCollision.cs [hunk 3]
```diff
@@ -2,20 +2,57 @@
...
+
+            CopyTransformHierachy(enemyRagdoll.transform, gameObject.transform);
+
             Destroy (gameObject);
-            Destroy (collision.gameObject);
+
+            enemyRagdoll.SetActive(true);
+
+            foreach (ContactPoint contactPoint in collision.contacts) {
+                Collider[] ragdollColliders = enemyRagdoll.GetComponentsInChildren<Collider>();
```
Comment (LaurieTrichet) [inline Assets/Scripts/EnemyBulletCollision.cs]: move this line outside the loop, not sure there is an optimization made by the compiler.

File: Assets/Scripts/ReduceLightIntensityOverTime.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using UnityEngine;
+
+public class ReduceLightIntensityOverTime : MonoBehaviour {
```
Comment (LaurieTrichet) [inline Assets/Scripts/ReduceLightIntensityOverTime.cs]: Replace this by an animation.

**Comment by rafaelgonzalez** [comment]
🆙

**Pull request merged**